gokay.go

```diff
@@ -49,12 +49,14 @@ func main() {
 
 	tempName := uuid.NewRandom().String()
 
-	tempDir := fmt.Sprintf("%s/%s", fileDir, tempName)
-	if err := os.Mkdir(tempDir, os.ModePerm); err != nil {
-		log.Fatalf("Error creating directory %v: %v\n", tempDir, err)
-	}
+	tempDir := fmt.Sprintf("%s/tmp/%s", fileDir, tempName)
 	tempFile := fmt.Sprintf("%s/%s.go", tempDir, tempName)
 
+	// os.Mkdir errors when the dir already exists, which is fine.
+	// Real errors will still be detected when we try use the actual tmpfile.
+	_ = os.Mkdir(fileDir+"/tmp", os.ModePerm)
+	_ = os.Mkdir(tempDir, os.ModePerm)
```

Comment:
Happy to put non-fatal logging back here if necessary, although imo it's not useful information.